### ISSUE TITLE
`stack ghci` can't find `file-embed` files when the package is in a subdirectory

### ISSUE THREAD

**Issue by mtolly** [commercialhaskell/stack#1057]
**Steps to reproduce**

1. Clone https://github.com/mtolly/stack-file-embed-bug -- repo has a top-level `stack.yaml`, a subdirectory with a package, and that package uses `file-embed` to reference a relative file.
2. `stack ghci`

**Expected**

The relative filepath is treated as relative to the package directory, *not* the top-level directory with the `stack.yaml` in it. `stack build` already does this correctly; only `stack ghci` fails.

**Actual**

```
╭─mtolly@metal  ~/g/stack-file-embed  <master*>
╰─$ stack build
the-package-0.1.0.0: configure
Configuring the-package-0.1.0.0...
the-package-0.1.0.0: build
Preprocessing executable 'the-package' for the-package-0.1.0.0...
[1 of 1] Compiling Main             ( src/Main.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/the-package/the-package-tmp/Main.o )
Linking .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/the-package/the-package ...
the-package-0.1.0.0: install
Installing executable(s) in
/Users/mtolly/git/stack-file-embed/.stack-work/install/x86_64-osx/lts-3.5/7.10.2/bin

╭─mtolly@metal  ~/g/stack-file-embed  <master*>
╰─$ stack exec the-package
This is the data file!

╭─mtolly@metal  ~/g/stack-file-embed  <master*>
╰─$ stack ghci
Using main module: Package `the-package' component exe:the-package with main-is file: /Users/mtolly/git/stack-file-embed/the-package/src/Main.hs
Configuring GHCi with the following packages: the-package
GHCi, version 7.10.2: http://www.haskell.org/ghc/  :? for help
[1 of 1] Compiling Main             ( /Users/mtolly/git/stack-file-embed/the-package/src/Main.hs, interpreted )

/Users/mtolly/git/stack-file-embed/the-package/src/Main.hs:8:19:
    Exception when trying to run compile-time code:
      data-file.txt: openBinaryFile: does not exist (No such file or directory)
    Code: embedFile "data-file.txt"
    In the splice: $(embedFile "data-file.txt")
Failed, modules loaded: none.
Prelude λ 
```

Of course, if I change it to `$(embedFile "the-package/data-file.txt")`, then `stack ghci` succeeds and `stack build` fails.

**Comment by luigy** [comment]
currently, running `stack ghci` from `the-package` directory should work
or if you add `:cd the-package` to a `.ghci` file

`stack build` runs the build with the target's package directory as the cwd (./the-package)

is there any case in which running `stack ghci target` with respect to the target's package directory might not be wanted as the default? maybe this would be a nice ux addition

**Comment by mtolly** [comment]
I think this is specifically about the directory that Template Haskell gets run in, not necessarily anything to do with the directory you are in once ghci is finished loading (which should probably still be whichever directory you run the `stack ghci` command from).

For example, what if there were 2 such subdirectory packages, each of which had its own TH-embedded files? Your suggested workarounds wouldn't work, since each one would need to have its TH run relative to its own directory.

**Comment by snoyberg** [comment]
The problem is that, with the current GHCi interface, there's no way to give it two different current directories for different sets of filepaths, meaning that there's no workaround. This is similar to the problem of having different default extensions set for different sub projects.

@chrisdone Can you confirm that this is in fact the case?

**Comment by snoyberg** [comment]
Closing as wontfix.

**Comment by mtolly** [comment]
This is a very annoying problem for me -- it breaks `stack ghci` completely in my repo with multiple packages that use `file-embed`. Has the missing functionality been mentioned over at GHC?

**Comment by snoyberg** [comment]
Not to my knowledge, can you open up a GHC Trac issue?
